core/dictgen/src/rootcling_impl.cxx

```diff
@@ -3218,7 +3218,7 @@ class tempFileNamesCatalog {
    tempFileNamesCatalog(): m_size(0), m_emptyString("") {};
 
    std::string getTmpFileName(const std::string &filename) {
-      return filename + "_tmp";
+      return filename + "_tmp_" + std::to_string(getpid());
```

Comment:
@Teemperor 
> We don't know if someone generates build files for something really obscure (e.g. Visual Studio or an old Make version) which chooses the target order in a different way

That's not the issue.

What you're saying is that build systems might trigger concurrent commands to generate the same target. That's horrendous. It's a bug. It would mean e.g. that the same library is linked twice in parallel. I buy your argument that this is a consequence of your CMake-based configuration to use `add_custom_command`.

I'd hope that we can *not* use `add_custom_command` and instead use something that allows CMake to construct a proper dependency tree.